rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf

```diff
@@ -436,12 +436,12 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_H
 #   \(.*\)
 #       Parentheses optionally containing function parameters
 #
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_FILENAME|ARGS_NAMES|ARGS|XML:/* "@rx \$+(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*|\s*{.+})(?:\s|\[.+\]|{.+}|/\*.*\*/|//.*|#.*)*\(.*\)" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_FILENAME|ARGS_NAMES|ARGS|XML:/* "@rx \$+(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*|{.+}).*\(.*\)" \
```

Comment:
Removing the comment markers forces you to match `.*` instead of `(?:\[.+\]|{.+}|/\*.*\*/|//.*|#.*)` (notice that I removed the `\s` from that expression). Does that really make sense? Yes, backtracking is better but the regex becomes much more open and prone to FPs. In combination with the removal of spaces you can easily produce FPS: `iget10$anhour(no,idon'tgetpaidforovertime)`. Besides, the regex is vulnerable to DOSing anyway.

I would compress spaces but not remove them, then you can match against single spaces at least.

Reply:
> Besides, the regex is vulnerable to DOSing anyway.

uhm why? @airween checked it but no redos were found anymore

> I would compress spaces but not remove them, then you can match against single spaces at least.

yep, I can try to fix it without removing whitespace. I'm going to send a new version tnx

Reply:
>uhm why? @airween checked it but no redos were found anymore

You're right. I had made a mistake when copying around different versions. Your version is fine.